index.js

```diff
@@ -1,55 +1,78 @@
-const {token} = require('./config.json');
-const Discord = require('discord.js');
-const client = new Discord.Client();
+const { token } = require('./config.json')
+const Discord = require('discord.js')
+const client = new Discord.Client()
 
 client.on('ready', () => {
-	console.log(`Logged in as ${client.user.tag}!`);
+    console.log(`Logged in as ${client.user.tag}!`)
 
-	//list servers bot is connected to
-	console.log("\nServers:");
-	client.guilds.cache.forEach((guild) => {
-		console.log(" - " + guild.name);
+    //list servers bot is connected to
+    console.log('\nServers:')
+    client.guilds.cache.forEach((guild) => {
+        console.log(' - ' + guild.name)
 
-		//list channels bot in on
-		guild.channels.cache.forEach((channel) => {
-			console.log(" -- " + channel.name + " (" + channel.type + ") - " + channel.id);
-		})
-	});
+        //list channels bot in on
+        guild.channels.cache.forEach((channel) => {
+            console.log(
+                ' -- ' +
+                    channel.name +
+                    ' (' +
+                    channel.type +
+                    ') - ' +
+                    channel.id
+            )
+        })
+    })
+})
 
- });
+client.on('message', (msg) => {
+    //console.log(msg);
+    var message = msg.content.toLowerCase()
+    var channelId = msg.channel.id
+    //console.log(msg.channel.id);
+    var channel = client.channels.cache.get(channelId)
+    if (message.startsWith('r!')) {
+        message = message.substring(2) //extracts the message without the "r!"
+        dicePattern = new RegExp('d+dd+/i')
+        if (message === 'beer') {
+            channel.send('yummy')
+        } else if (message === 'flops') {
+            channel.send('F')
+            channel.send(':clap::clap:    :clap::clap:')
+            channel.send('L')
+            channel.send(':clap::clap:    :clap::clap:')
+            channel.send('O')
+            channel.send(':clap::clap:    :clap::clap:')
+            channel.send('P')
+            channel.send(':clap::clap:    :clap::clap:')
+			channel.send('Flopsssssss')	
+        } else if (message.match(dicePattern)) { // checks if message is formatted like [numDice]d[numSides]
+            var xy = message.split('d')
+            var numDice = parseInt(xy[0])
+            var numSides = parseInt(xy[1])
 
-client.on('message', msg => {
-	//console.log(msg);
-	var message = msg.content.toLowerCase();
-	var channelId = msg.channel.id;
-	//console.log(msg.channel.id);
-	var channel = client.channels.cache.get(channelId);
-	if (message.startsWith('r!')){
-		 message = message.substring(2); //extracts the message without the "r!"
+            for (let i = 0; i < numDice; i++) {
```

Comment:
The way it behaves right now, it looks like it won't roll the dice at the same time, which is not really a desired effect. If for example I roll 100d6 it will be spamming for a while.

Reply:
Considering the next comment about setTimeout, it would not spam for a while but send 100 messages at once. I suggest having a loop concatenate a string and then sending that string.